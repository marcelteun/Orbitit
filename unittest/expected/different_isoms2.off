OFF
#
# SimpleShape
#
# file generated with python script by Marcel Tunnissen
# Vertices Faces Edges
12 12 18
# Vertices
1.060660171779821 0.353553390593274 1
-1.414213562373095 0 1
0 1.414213562373095 -1
0 -1.414213562373095 -1
-1.060660171779821 0.353553390593274 1
1.414213562373095 0 1
0 1.414213562373095 -1
0 -1.414213562373095 -1
0.353553390593274 -1 1.060660171779821
0 -1 -1.414213562373095
1.414213562373095 1 0
-1.414213562373095 1 0
# Sides and colours
3  0 2 1  254 214 0
3  0 1 3  254 214 0
3  0 3 2  254 214 0
3  1 2 3  254 214 0
3  4 6 5  33 138 33
3  4 5 7  33 138 33
3  4 7 6  33 138 33
3  5 6 7  33 138 33
3  8 10 9  138 0 0
3  8 9 11  138 0 0
3  8 11 10  138 0 0
3  9 10 11  138 0 0
# END
