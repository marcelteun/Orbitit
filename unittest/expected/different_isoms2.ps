%!PS-Adobe-2.0
%%Title: SimpleShape
%%Creator: PS.py written by marcelteun
%%Pages: 1 1
%%BoundingBox: 0 0 595 842
%%EndComments
%%BeginSetup
%%EndSetup
%%Page: 1 1
%%BeginPage
10.000000 10.000000 translate
/vertices [
  [1.1956601 -0.5914462] % 0
  [-1.2792136 -0.9449996] % 1
  [-0.0251429 1.5902137] % 2
  [-0.0251429 1.5902137] % 3
  [0.1349999 -0.7429691] % 4
  [0.1349999 -0.7429691] % 5
  [-0.8385134 -0.6540366] % 6
  [-0.8385134 -0.6540366] % 7
  [-0.0251429 1.5902137] % 8
  [0.6375558 0.4059251] % 9
  [-0.8080142 0.0075711] % 10
  [-0.5656027 0.497627] % 11
  [0.5488091 0.5645216] % 12
] def
/faces [
  [0 2 1][3 4][5 6][7 8][9 10][11 12]
] def
/bbox [-1.2792136 -0.9449996 1.1956601 1.5902137] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
128.743687 0 translate
/vertices [
  [0.9608439 1.0522675] % 0
  [-1.5140299 0.6987141] % 1
  [0.0361249 -1.6670881] % 2
  [-0.0998163 0.9007446] % 3
  [0.8713549 0.789104] % 4
  [0.0361249 -1.6670881] % 5
  [-0.0998163 0.9007446] % 6
  [0.6345855 0.0928272] % 7
  [0.1319729 -1.0198614] % 8
  [0.1319729 -1.0198614] % 9
  [-0.4601111 0.353453] % 10
  [-0.4601111 0.353453] % 11
  [0.7539681 0.4439003] % 12
] def
/faces [
  [0 1 2][3 4][5 6][7 8][9 10][11 12]
] def
/bbox [-1.5140299 -1.6670881 0.9608439 1.0522675] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
128.743687 0 translate
/vertices [
  [-1.3803941 0.3535534] % 0
  [0.8834522 1.4142136] % 1
  [0.8834522 -1.4142136] % 2
  [-1.1613122 0.1824792] % 3
  [0.8834522 1.4142136] % 4
  [0.7105893 -1] % 5
  [-0.5816661 -0.2701484] % 6
  [0.8834522 -0.9267767] % 7
  [0.7105893 -1] % 8
  [-0.8739322 -0.0419269] % 9
  [-0.3454503 0.8384466] % 10
  [-0.1808792 0.9155516] % 11
  [0.8834522 -0.4142136] % 12
] def
/faces [
  [0 2 1][3 4][5 6][7 8][9 10][11 12]
] def
/bbox [-1.3803941 -1.4142136 0.8834522 1.4142136] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
118.192314 0 translate
/vertices [
  [1.6329932 0] % 0
  [-0.8164966 1.4142136] % 1
  [-0.8164966 -1.4142136] % 2
  [0.773079 0.4383064] % 3
  [-0.8164966 -0.9267767] % 4
  [0.7126319 0.5313708] % 5
  [0.773079 0.4383064] % 6
  [-0.8164966 -0.4142136] % 7
  [0.2391463 0.8047379] % 8
] def
/faces [
  [0 1 2][3 4][5 6][7 8]
] def
/bbox [-0.8164966 -1.4142136 1.6329932 1.4142136] def
/scalingSize 50 def
.1 setlinewidth

/dx bbox 2 get bbox 0 get sub scalingSize mul def
/dy bbox 3 get bbox 1 get sub scalingSize mul def

gsave
  bbox 0 get scalingSize mul neg
  bbox 1 get scalingSize mul neg
  translate
  faces aload length { %repeat
    dup 0 get
    vertices exch get
    aload pop
    scalingSize mul exch
    scalingSize mul exch
    moveto
    aload length 1 sub { %repeat
      vertices exch get
      aload pop
      scalingSize mul exch
      scalingSize mul exch
      lineto
    } repeat
    pop
    closepath stroke
  } repeat


grestore
127.474487 0 translate

showpage
%%EndPage

%%Tailer
%%EOF
