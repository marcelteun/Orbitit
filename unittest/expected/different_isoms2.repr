orbitit.geom_3d.SymmetricShape(
    vs=[
        orbitit.geomtypes.Vec3([1, -0.5, 1]),
        orbitit.geomtypes.Vec3([-1, 1, 1]),
        orbitit.geomtypes.Vec3([1, 1, -1]),
        orbitit.geomtypes.Vec3([-1, -1, -1])
    ],
    fs=[
        [0, 2, 1],
        [0, 1, 3],
        [0, 3, 2],
        [1, 2, 3]
    ],
    Es=[0, 2, 1, 2, 0, 1, 1, 3, 0, 3, 2, 3],
    colors=[
        [0.996094, 0.839844, 0.0],
        [0.132812, 0.542969, 0.132812],
        [0.542969, 0.0, 0.0]
    ],
    isometries=[
        orbitit.geomtypes.Rot3((
            orbitit.geomtypes.Quat([1, 0, 0, 0]),
            orbitit.geomtypes.Quat([1, 0, 0, 0]),
        )),
        orbitit.geomtypes.Refl3((
            orbitit.geomtypes.Quat([0, 1, 0, 0]),
            orbitit.geomtypes.Quat([0, 1, 0, 0]),
        )),
        orbitit.geomtypes.RotInv3((
            orbitit.geomtypes.Quat([-0.5, 0.5, -0.5, -0.5]),
            orbitit.geomtypes.Quat([0.5, 0.5, -0.5, -0.5]),
        ))
    ],
    name='different_isoms2',
    orientation=orbitit.geomtypes.Rot3((
        orbitit.geomtypes.Quat([0.923879532511, 0, 0, 0.382683432365]),
        orbitit.geomtypes.Quat([0.923879532511, 0, 0, -0.382683432365]),
    ))
)
