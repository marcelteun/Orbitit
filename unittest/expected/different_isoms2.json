{"class": "symmetric_shape", "data": {"cols": [[0.996094, 0.839844, 0], [0.132812, 0.542969, 0.132812], [0.542969, 0, 0]], "fs": [[0, 2, 1], [0, 1, 3], [0, 3, 2], [1, 2, 3]], "isometries": [{"class": "transform_3d", "data": [{"class": "quaternion", "data": [1, 0, 0, 0]}, {"class": "quaternion", "data": [1, 0, 0, 0]}]}, {"class": "transform_3d", "data": [{"class": "quaternion", "data": [0, 1, 0, 0]}, {"class": "quaternion", "data": [0, 1, 0, 0]}]}, {"class": "transform_3d", "data": [{"class": "quaternion", "data": [-0.5, 0.5, -0.5, -0.5]}, {"class": "quaternion", "data": [0.5, 0.5, -0.5, -0.5]}]}], "name": "different_isoms2", "orientation": {"class": "transform_3d", "data": [{"class": "quaternion", "data": [0.923879532511, 0, 0, 0.382683432365]}, {"class": "quaternion", "data": [0.923879532511, 0, 0, -0.382683432365]}]}, "vs": [[1, -0.5, 1], [-1, 1, 1], [1, 1, -1], [-1, -1, -1]]}}